qt/scientific_interfaces/Muon/ALCDataLoadingView.cpp

```diff
@@ -421,4 +442,43 @@ std::string ALCDataLoadingView::getAlphaValue() const {
 
 void ALCDataLoadingView::showAlphaMessage(const bool alpha) { m_ui.alphaMessage->setVisible(alpha); }
 
+std::shared_ptr<MantidQt::MantidWidgets::MuonPeriodInfo> ALCDataLoadingView::getPeriodInfo() { return m_periodInfo; }
+
+QFileSystemWatcher *ALCDataLoadingView::getFileSystemWatcher() { return m_watcher; }
+
+QTimer *ALCDataLoadingView::getTimer() { return m_timer; }
+
+void ALCDataLoadingView::handleStartWatching(bool watch) {
+  if (watch) {
+    // Get path to watch and add to watcher
+    const auto path = getPath();
+    m_watcher->addPath(QString::fromStdString(path));
+    // start a timer that executes every second
+    getTimer()->start(1000);
+
+  } else {
+    // Check if watcher has a directory, then remove all
+    if (!m_watcher->directories().empty()) {
+      m_watcher->removePaths(m_watcher->directories());
+    }
+    // Stop timer
+    getTimer()->stop();
+
+    m_presenter->handleWatcherStopped();
+  }
+}
```

Comment:
As a final question, this is the only sort of logic present in the view, but I put it here because both `m_wacher` and `m_timer` are QObjects

Reply:
I think that reason is fair enough, so I'm happy for it to stay as it is!